sandbox/errorHandling.lua

```diff
@@ -124,6 +120,35 @@ function errorHandling:_PlayerSpaceship()
 	end
 end
 
+function errorHandling:_CpuShip()
+	local create = CpuShip
+	return function()
+		local ship = create()
+		self:_AddSpaceShipErrorHandling(ship)
+		return ship
+	end
+end
+
+function errorHandling:_SpaceStation()
+	local create = SpaceStation
+	return function()
+		local station = create()
+		self:_AddShipTemplateBasedObjectErrorHandling(station)
+		return station
+	end
+end
+
+function errorHandling:_AddShipTemplateBasedObjectErrorHandling(ship)
+	ship.onTakingDamage = self:_autoWrapArgX(ship.onTakingDamage,2)
+	ship.onDestruction = self:_autoWrapArgX(ship.onDestruction,2)

```

Comment:
There's also an onDestroyed in addition to onDestruction where onDestruction is described as being destroyed by damage. I'm not sure exactly what other differences there might be.

Reply:
onDestroyed is part of SpaceObject - I think it may be an effect of some non damage effects (center of a black hole), I have on a list to investigate it as a single mine hitting a PlayerShip was inconsistant as to if onDestruction was called

At a minium it needs better documenting

Regardless SpaceObject:onDestroyed is on the todo list near the earlier part of the file, so I will wrap round to it when I have time